virtcontainers/implementation.go

```diff
@@ -18,16 +18,22 @@ import (
 
 // VCImpl is the official virtcontainers function of the same name.
 type VCImpl struct {
+	factory Factory
 }
 
 // SetLogger implements the VC function of the same name.
 func (impl *VCImpl) SetLogger(logger logrus.FieldLogger) {
 	SetLogger(logger)
 }
 
+// SetFactory implements the VC function of the same name.
+func (impl *VCImpl) SetFactory(factory Factory) {
```

Comment:
You need the mock implementation to implement this new API.

Reply:
The mock implementation has `SetFactory`.